src/Microsoft.AspNetCore.Blazor/MarkupString.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Blazor
+{
+    /// <summary>
+    /// A string value that can be rendered as markup such as HTML.
+    /// </summary>
+    public struct MarkupString
+    {
+        private readonly string _value;
+
+        /// <summary>
+        /// Constructs an instance of <see cref="MarkupString"/>.
+        /// </summary>
+        /// <param name="value">The value for the new instance.</param>
+        public MarkupString(string value)
+        {
+            _value = value ?? string.Empty;
```

Comment:
Why this? Is it ever meaningful to have null markup string?

For instance the following is true. `((MarkupString)null).ToString() == ""`

Reply:
Good point.

I've now changed it so that `MarkupString` can hold the value `null`.

 * If you `.ToString()` such an instance, you get `string.Empty` just like HtmlAbstractions' `HtmlString`.
 * If you `renderTreeBuilder.AddContent` such an instance, it creates a markup frame whose value is `string.Empty`, just like if you `AddContent((string)null)`.

So the net effect is that nulls continue to be treated like `string.Empty`, but now it's possible to observe that `null` was stored in the `MarkupString` to avoid losing that info.